pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1080,7 +1186,7 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *v1.Service
 			_, err := v2pools.CreateMember(lbaas.network, pool.ID, v2pools.CreateMemberOpts{
 				Address:      addr,
 				ProtocolPort: int(port.NodePort),
-				SubnetID:     lbaas.opts.SubnetId,
+				SubnetID:     getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerSubnetID, lbaas.opts.SubnetId, lbaas.opts.EnableAnnotationSubnetId),
```

Comment:
Currently there are three ways to get subnet-id:
1.lbaas.opts.SubnetId
2.ServiceAnnotationLoadBalancerSubnetID
3.autoprobe subnet-id by "getSubnetIDForLB()"
You should also consider 3. If 1 and 2 are empty, we can use 3.